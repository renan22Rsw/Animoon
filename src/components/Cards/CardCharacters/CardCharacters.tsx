import Image from "next/image";
import Link from "next/link";
import React from "react";

interface CardCharactersProps {
  id: number;
  name: string;
  image: string;
  role: string;
  voices: {
    id: number;
    name: string;
    image: string;
    language: string;
  }[];
}

const CardCharacters = ({
  name,
  image,
  role,
  voices,
  id,
}: CardCharactersProps) => {
  return (
    <div className="flex justify-between bg-[#1C1C1C] h-[80px] w-[350px] m-4 lg:w-[381px] 2xl:w-[375px]">
      <Link
        href={`/characters/${id}`}
        className="flex items-center text-xs"
        data-testid="character-link"
      >
        <Image
          className="h-25 w-[60px] rounded-sm 2xl:h-[80px] 2xl:w-[70px] "
          src={image}
          alt="Card-image-character"
          width={60}
          height={60}
          priority={true}
          style={{ width: "54px", height: "auto" }}
        />
        <span className="p-2 grid items-center w-[180px]  gap-y-6">
          <h4>{name}</h4>
          <p>{role.charAt(0).toUpperCase() + role.slice(1).toLowerCase()}</p>
        </span>
      </Link>

      <Link
        href={`/staffs/${voices[0]?.id}`}
        className="flex items-center text-xs"
        data-testid="voice-link"
      >
        <span className=" p-2 grid  items-center w-[180px] gap-y-6 text-end">
          <h4>{voices[0]?.name}</h4>
          <p>{voices[0]?.language}</p>
        </span>
        {voices[0]?.image && (
          <Image
            className="h-20 w-[60px] rounded-sm 2xl:h-[80px] 2xl:w-[70px]"
            src={voices[0]?.image}
            alt="Card-image-voices"
            width={60}
            height={60}
            priority={true}
          />
        )}
      </Link>
    </div>
  );
};

export default CardCharacters;
