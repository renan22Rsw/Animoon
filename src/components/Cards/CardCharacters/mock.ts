interface mockTypes {
  id: number;
  name: string;
  image: string;
  role: string;
  voices: {
    id: number;
    name: string;
    image: string;
    language: string;
  }[];
}

export const mock: mockTypes = {
  id: 1,
  name: "Naruto",
  image: "https://example.com/character-image.jpg",
  role: "Main",
  voices: [
    {
      id: 1,
      name: "Junko Takeuch",
      image: "https://example.com/voices-image.jpg",
      language: "Japanease",
    },
  ],
};
