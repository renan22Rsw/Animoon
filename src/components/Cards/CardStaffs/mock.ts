interface mockTypes {
  id: number;
  name: string;
  image: string;
  occupation: string;
}

export const mock: mockTypes = {
  id: 1,
  name: "Masashi Kishimoto",
  image: "https://example.com/staff-image.jpg",
  occupation: "Original Creator",
};
