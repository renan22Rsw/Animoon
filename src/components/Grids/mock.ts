interface mockAnimesAndMangas {
  id: number;
  title: {
    romaji: string;
  };
  coverImage: {
    large: string;
  };
}

interface mockCharacter {
  id: number;
  name: {
    full: string;
  };
  image: {
    large: string;
  };
}

export const mockAnimesAndMangasMainPage: mockAnimesAndMangas = {
  id: 1,
  title: { romaji: "Akame ga kill" },
  coverImage: { large: "https://example.com/anime-image.jpg" },
};

export const mockCharacterMainPage: mockCharacter = {
  id: 2,
  name: { full: "Akame" },
  image: { large: "https://example.com/character-image.jpg" },
};
