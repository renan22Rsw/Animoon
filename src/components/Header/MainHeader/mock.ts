interface mockTypes {
  title: string;
  description: string;
  coverImage: string;
}

export const mock: mockTypes = {
  title: "Naruto",
  description:
    "Naruto Uzumaki, a hyperactive and knuckle-headed ninja, lives in Konohagakure, the Hidden Leaf village. Moments prior to his birth, a huge demon known as the Kyuubi, the Nine-tailed Fox, attacked Konohagakure and wreaked havoc. In order to put an end to the Kyuubi's rampage, the leader of the village, the 4th Hokage, sacrificed his life and sealed the monstrous beast inside the newborn Naruto.",
  coverImage: "https://example.com/image.jpg",
};
