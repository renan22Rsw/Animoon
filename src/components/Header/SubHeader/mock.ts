interface mockTypes {
  name: string;
  nativeName: string;
  alternativeName?: string[];
  age: string;
  image: string;
  gender: string;
  month: number;
  day: number;
  bloodType: string;
  hometown: string;
  yearsActive?: number[];
  description: string;
}

export const mock: mockTypes = {
  name: "Naruto",
  nativeName: "Naruto Uzumaki",
  alternativeName: ["Nine-Tails Jinchuuriki", " Number One Unpredictable"],
  age: "17",
  image: "http://example.com/image.jpg",
  gender: "Male",
  month: 10,
  day: 10,
  bloodType: "B",
  hometown: "Saitama Prefecture, Japan",
  yearsActive: [1996],
  description: "Born in Konohagakure, a ninja village hidden in the leaves",
};
