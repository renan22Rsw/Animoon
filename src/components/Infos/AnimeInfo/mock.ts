interface mockTypes {
  format: string;
  duration: number;
  status: string;
  seasonYear: number;
  season: string;
  averageScore: number;
  meanScore: number;
  popularity: number;
  favourites: number;
  source: string;
  genres: string[];
}

export const mock: mockTypes = {
  duration: 24,
  averageScore: 85,
  meanScore: 88,
  favourites: 500,
  format: "TV",
  genres: ["comedy", "horror"],
  popularity: 1000,
  season: "winter",
  seasonYear: 2024,
  source: "Light Novel",
  status: "Airing",
};
