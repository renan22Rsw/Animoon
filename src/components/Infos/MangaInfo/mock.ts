interface mockTypes {
  format: string;
  status: string;
  averageScore: number;
  meanScore: number;
  popularity: number;
  favourites: number;
  source: string;
  genres: string[];
}

export const mock: mockTypes = {
  averageScore: 80,
  meanScore: 85,
  favourites: 1200,
  format: "Manga",
  genres: ["action", "adventure"],
  popularity: 1000,
  source: "Original",
  status: "Releasing",
};
