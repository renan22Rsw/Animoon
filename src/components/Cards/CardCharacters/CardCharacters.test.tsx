import { render, screen } from "@testing-library/react";
import CardCharacters from "./CardCharacters";
import { mock } from "./mock";

describe("CardCharacters component", () => {
  it("should render CardCharacters Images", () => {
    render(<CardCharacters {...mock} />);

    const characterImage = screen.getByAltText("Card-image-character");
    expect(characterImage).toBeInTheDocument();
    expect(characterImage).toHaveAttribute("width", "60");
    expect(characterImage).toHaveAttribute("height", "60");
    expect(characterImage).toHaveAttribute("fetchpriority", "high");

    const voiceImage = screen.getByAltText("Card-image-voices");
    expect(voiceImage).toBeInTheDocument();
    expect(voiceImage).toHaveAttribute("width", "60");
    expect(voiceImage).toHaveAttribute("height", "60");
    expect(voiceImage).toHaveAttribute("fetchpriority", "high");
  });

  it("should render CardCharacters Links", () => {
    render(<CardCharacters {...mock} />);

    const characterLink = screen.getByTestId("character-link");
    expect(characterLink).toBeInTheDocument();
    expect(characterLink).toHaveAttribute("href", `/characters/${mock.id}`);

    const voiceLink = screen.getByTestId("voice-link");
    expect(voiceLink).toBeInTheDocument();
    expect(voiceLink).toHaveAttribute("href", `/staffs/${mock.id}`);
  });

  it("should render titles of CardCharacters", () => {
    render(<CardCharacters {...mock} />);

    expect(screen.getByText(mock.name)).toBeInTheDocument();
    expect(screen.getByText(mock.role)).toBeInTheDocument();
    expect(screen.getByText(mock.voices[0].name)).toBeInTheDocument();
    expect(screen.getByText(mock.voices[0].language)).toBeInTheDocument();
  });

  it("should match a snapshot", () => {
    const { container } = render(<CardCharacters {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
