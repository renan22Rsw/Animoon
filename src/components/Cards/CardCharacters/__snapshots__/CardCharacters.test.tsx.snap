// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCharacters component should match a snapshot 1`] = `
<div>
  <div
    class="flex justify-between bg-[#1C1C1C] h-[80px] w-[350px] m-4 lg:w-[381px] 2xl:w-[375px]"
  >
    <a
      class="flex items-center text-xs"
      data-testid="character-link"
      href="/characters/1"
    >
      <img
        alt="Card-image-character"
        class="h-25 w-[60px] rounded-sm 2xl:h-[80px] 2xl:w-[70px] "
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="60"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Fcharacter-image.jpg&w=128&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fcharacter-image.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fcharacter-image.jpg&w=128&q=75 2x"
        style="color: transparent; width: 54px; height: auto;"
        width="60"
      />
      <span
        class="p-2 grid items-center w-[180px]  gap-y-6"
      >
        <h4>
          Naruto
        </h4>
        <p>
          Main
        </p>
      </span>
    </a>
    <a
      class="flex items-center text-xs"
      data-testid="voice-link"
      href="/staffs/1"
    >
      <span
        class=" p-2 grid  items-center w-[180px] gap-y-6 text-end"
      >
        <h4>
          Junko Takeuch
        </h4>
        <p>
          Japanease
        </p>
      </span>
      <img
        alt="Card-image-voices"
        class="h-20 w-[60px] rounded-sm 2xl:h-[80px] 2xl:w-[70px]"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="60"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Fvoices-image.jpg&w=128&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fvoices-image.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fvoices-image.jpg&w=128&q=75 2x"
        style="color: transparent;"
        width="60"
      />
    </a>
  </div>
</div>
`;
